apps/flink.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+)
+
+type MetricsReceiverFlink struct {
+	confgenerator.ConfigComponent       `yaml:",inline"`
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+	Endpoint                            string `yaml:"endpoint" validate:"omitempty,url,startswith=http:"`
+}
+
+func (MetricsReceiverFlink) Type() string {
+	return "flinkmetrics"
```

Comment:
Can we rename the metrics type to just `flink`?

Reply:
updated. changed to flink